content/general/concepts/memoization/memoization.md

```diff
@@ -0,0 +1,23 @@
+---
+Title: 'Memoization'
+Description: 'Memoization is a technique that improves the speed of a program by storing any new inputs for future reference, if needed.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Algorithms'
+  - 'Conceptual'
+  - 'Dynamic Programming'
+CatalogContent:
+  - 'learn-how-to-code'
+  - 'paths/computer-science'
+---
+
+Memoization is a technique that improves the speed of a program by storing any new inputs for future reference, if needed. This cuts down on any repeated computations in the program where the same inputs are being used to return the same result.
+
+This technique gets its name from the Latin word, "memorandum", that means "to be remembered." Memoization can be applied to many contexts in computer programming, including:
+
+- Making calls to recursive or pure function where some calculations use the same inputs and return the same outputs.
+- Fetching from the server with the same base API endpoints.
+
+Memoization can be implemented in most programming languages, including [JavaScript](https://www.codecademy.com/resources/docs/javascript/memoization) and [Python](https://www.codecademy.com/resources/docs/python/memoization). However, while memoization saves performance time it comes with a cost in memory space since it is caching results that were already calculated.
```

Comment:
```suggestion
Memoization can be implemented in most programming languages, including [JavaScript](https://www.codecademy.com/resources/docs/javascript/memoization) and [Python](https://www.codecademy.com/resources/docs/python/memoization). However, while memoization saves performance time, it comes with a cost in memory space since it is caching results that were already calculated.
```